docs/rationale.md

```diff
@@ -23,4 +23,69 @@ try (Scope scope = TracingContextUtils.currentContextWith(span)) {
 ```
 
 It would not be possible to call `recordException` if `span` was also using try-with-resources.
-Because this is a common usage for spans, we do not support try-with-resources.
\ No newline at end of file
+Because this is a common usage for spans, we do not support try-with-resources.
+
+
+## Versioning and Releases
+
+### Assumptions
+
+- This project uses semver v2, as does the rest of OpenTelemetry.
+
+### Goals
+
+- API Stability: 
+    - Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will 
+function, *with no recompilation required*, with any SDK that has the same major version, and equal or greater minor or patch version. 
+    - For example, libraries that are instrumented with `opentelemetry-api-trace:1.0.1` will function with 
```

Comment:
Looking through https://github.com/open-telemetry/oteps/pull/143/files I couldn't find a point on this so I guess it's just #2203 :)

Given using a bom is easy and we can strongly encourage it, I don't know if we need to require this

Reply:
What are you saying we don't need to require?  Shouldn't an 1.x API work with any 1.y SDK as long as y >= x?

Reply:
But, if there is a library that is instrumented with API v. 1.0.0, that would mean you can't use SDK v. 1.1.0 with it? Doesn't that defeat the entire purpose of ABI compatibility, let alone API compatibility? This seems to directly contradict what you're saying about SDK ABI requirements! 

Or else, we're talking at cross purposes about something!

Reply:
@anuraaga this statement, as written here, MUST be true. If my app has a dependency on a library, which is instrumented (meaning was compiled against) API v1.0.1, then it MUST work if offered SDK v1.11.33 during runtime.

I think what you want to fix in writing is "should we allow having different versions of API and SDK during _runtime_". E.g. is it OK for application developer to specify in their build.gradle SDK v1.11.33 and API v1.0.1? I agree with Anuraag here that we should discourage that.

But! What if an application developer has SDK v1.11.33 and some library depends on API v1.20. What will SDK do?

Reply:
What will it do will obviously depend on the nature of the changes. If they happen to be backward compatible, then things should work. And, if they aren't, then probably all hell will break loose and NoClassDefFoundErrors will abound at runtime, unfortunately. 

Can we maybe detect versions at startup and fail fast if there are likely to be compatibility issues?